test/AsioHelpersTest.cpp

```diff
@@ -4,269 +4,108 @@
 
 #include <gtest/gtest.h>
 
-#include <boost/asio/bind_executor.hpp>
-#include <boost/asio/co_spawn.hpp>
-#include <boost/asio/deadline_timer.hpp>
-#include <boost/asio/detached.hpp>
-#include <boost/asio/experimental/awaitable_operators.hpp>
-#include <boost/asio/io_context.hpp>
-#include <boost/asio/steady_timer.hpp>
-#include <boost/asio/strand.hpp>
-
 #include "util/AsioHelpers.h"
-
-namespace net = boost::asio;
-
-using ad_utility::resumeOnOriginalExecutor;
-using namespace boost::asio::experimental::awaitable_operators;
-
-TEST(AsioHelpers, resumeOnOriginalExecutor) {
-  net::io_context ioContext{};
-  auto strand1 = net::make_strand(ioContext);
-  auto strand2 = net::make_strand(ioContext);
-
-  uint32_t sanityCounter = 0;
-
-  auto innerAwaitable = [&sanityCounter, strand1,
-                         strand2]() -> net::awaitable<int> {
-    co_await net::post(net::bind_executor(strand2, net::use_awaitable));
-    // sanity check
-    EXPECT_FALSE(strand1.running_in_this_thread());
-    EXPECT_TRUE(strand2.running_in_this_thread());
-    sanityCounter++;
-    co_return 1337;
-  };
-
-  auto outerAwaitable = [&sanityCounter, &innerAwaitable, strand1,
-                         strand2]() -> net::awaitable<void> {
-    // Sanity check
-    EXPECT_TRUE(strand1.running_in_this_thread());
-    EXPECT_FALSE(strand2.running_in_this_thread());
-    auto value = co_await resumeOnOriginalExecutor(innerAwaitable());
-    // Verify we're back on the same strand
-    EXPECT_TRUE(strand1.running_in_this_thread());
-    EXPECT_FALSE(strand2.running_in_this_thread());
-    EXPECT_EQ(value, 1337);
-    sanityCounter++;
-  };
-
-  net::co_spawn(strand1, outerAwaitable(), net::detached);
-
-  ioContext.run();
-
-  EXPECT_EQ(sanityCounter, 2);
+#include "util/http/beast.h"
+
+using namespace ad_utility;
+// _________________________________________________________________________
+TEST(AsioHelpers, runFunctionOnExecutorVoid) {
+  net::io_context ctx;
+  bool a = false;
+  auto workGuard = net::executor_work_guard<decltype(ctx.get_executor())>{
```

Comment:
Maybe? Of course the same would apply to all the other usages, it could also declared globally within this cpp file.
```suggestion
  auto workGuard = net::executor_work_guard<net::io_context::executor_type>{
```